sklearn/tree/_tree.pyx

```diff
@@ -438,7 +438,7 @@ cdef class BestFirstTreeBuilder(TreeBuilder):
         n_node_samples = end - start
         is_leaf = ((depth > self.max_depth) or
                    (n_node_samples < self.min_samples_split) or
-                   (n_node_samples < 2 * self.min_samples_leaf) or
+                   (n_node_samples < self.min_samples_leaf) or
```

Comment:
Yes, that's what I've been saying in the  other issue: that condition is completely broken. It's setting a leaf in situations that violate the condition that the parameter is meant to ensure. That's why I assume that the condition is never true: all management of `min_weight_leaf` should happen in `Splitter`.